### PR TITLE
add warning when 0 EOG/ECG-related IC detected ?

### LINKED ISSUE #347: Warning when 0 EOG-related IC detected ?
During the preprocessing step of the pipeline, I obtained:

`12:09:41 [Step-04][sub-403] Performing automated EOG artifact detection …`
`12:09:41 [Step-04][sub-403] Detected 0 EOG-related ICs in 18 EOG epochs.`
`12:10:46 [Step-04][sub-403] Saving ICA solution and detected artifacts to disk.`

I think it would be great if a warning appeared saying:

"Warning: zero EOG-related IC was detected, the EOG epochs will not be corrected.
Checking that EOG epochs are not superposed with actual task epochs...
There are 3 superpositions between task epochs and OEG epochs. Marking those 3 task epochs as rejected."

Or maybe it could just say:
"Warning: zero EOG-related IC was detected, this is highly suspicious. A manual check is required." 

What do you think?

### PR DESCRIPTION
**Pull request by crsegerie**
Closes #347

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: docs/source/changes.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ authors:
 - `bids_root` and `deriv_root` are now converted to absolute paths to avoid
   running into issues caused by relative path specifications.
   ({{ gh(322) }} by {{ authors.hoechenberger }})
+- Solve issue 337: Warning when 0 EOG-related IC detected.
```
Comment (hoechenberger) [inline docs/source/changes.md:42]:
```suggestion
- Warn if using ICA and  no EOG-related ICs were detected.
```

File: scripts/preprocessing/04a-run_ica.py [hunk 2]
```diff
@@ -139,6 +139,13 @@ def detect_ecg_artifacts(ica, raw, subject, session, report):
                f'{len(ecg_epochs)} ECG epochs.')
         logger.info(gen_log_message(message=msg, step=4, subject=subject,
                                     session=session))
+        if len(ecg_inds) == 0:
+            warn = (f'Zero ECG-related IC was detected, '
+                    f'this is highly suspicious.'
+                    f'A manual check is required.'
+                    f'You can try to lower ica_ctps_ecg_threshold.')
```
Comment (hoechenberger) [inline scripts/preprocessing/04a-run_ica.py:146]:
```suggestion
            warn = (f'No ECG-related ICs detected, '
                    f'this is highly suspicious. '
                    f'A manual check is suggested. '
                    f'You can try to lower "ica_ctps_ecg_threshold".')
```

File: scripts/preprocessing/04a-run_ica.py [hunk 3]
```diff
@@ -139,6 +139,14 @@ def detect_ecg_artifacts(ica, raw, subject, session, report):
                f'{len(ecg_epochs)} ECG epochs.')
         logger.info(gen_log_message(message=msg, step=4, subject=subject,
                                     session=session))
+        if len(ecg_inds) == 0:
+            warn = (f'Zero ECG-related IC was detected, '
+                    f'this is highly suspicious.'
```
Comment (agramfort) [inline scripts/preprocessing/04a-run_ica.py:144]:
```suggestion
                    f'this is highly suspicious. '
```

File: scripts/preprocessing/04a-run_ica.py [hunk 4]
```diff
@@ -139,6 +139,14 @@ def detect_ecg_artifacts(ica, raw, subject, session, report):
...
+                    f'A manual check is required.'
```
Comment (agramfort) [inline scripts/preprocessing/04a-run_ica.py:145]:
```suggestion
                    f'A manual check is required. '
```

File: scripts/preprocessing/04a-run_ica.py [hunk 5]
```diff
@@ -203,6 +211,16 @@ def detect_eog_artifacts(ica, raw, subject, session, report):
                f'{len(eog_epochs)} EOG epochs.')
         logger.info(gen_log_message(message=msg, step=4, subject=subject,
                                     session=session))
+        if len(eog_inds) == 0:
+            warn = (f'Zero EOG-related IC was detected, '
+                    f'this is highly suspicious.'
+                    f'A manual check is required.'
+                    f'You can either lower ica_eog_threshold or check'
+                    f' that EOG epochs are not superposed '
+                    f'with actual task epochs')
```
Comment (agramfort) [inline scripts/preprocessing/04a-run_ica.py:220]:
```suggestion
            warn = (f'Zero EOG-related IC was detected, '
                    f'this is highly suspicious. '
                    f'A manual check is required. '
                    f'You can either lower ica_eog_threshold or check'
                    f' that EOG epochs are not superposed '
                    f'with actual task epochs')
```

File: docs/source/changes.md [hunk 6]
```diff
@@ -39,6 +39,8 @@ authors:
...
+  ({{ gh(?) }} by {{ authors.crsegerie }})
```
Comment (hoechenberger) [inline docs/source/changes.md:43]:
```suggestion
  ({{ gh(351) }} by {{ authors.crsegerie }})
```

File: scripts/preprocessing/04a-run_ica.py [hunk 7]
```diff
@@ -203,6 +210,15 @@ def detect_eog_artifacts(ica, raw, subject, session, report):
...
```
[the remark this answers is not in the document]
Comment (crsegerie) [reply scripts/preprocessing/04a-run_ica.py]: That means that if all the blinks are between the tasks and not during the task, this is not really a problem

**Review by agramfort — APPROVED** [review]

**Review by hoechenberger — COMMENTED** [review]
Small nitpick but otherwise looking good to me!

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
LGTM one CI goes green. Thanks @crsegerie!

File: scripts/preprocessing/04a-run_ica.py [hunk 8]
```diff
@@ -203,6 +211,16 @@ def detect_eog_artifacts(ica, raw, subject, session, report):
                f'{len(eog_epochs)} EOG epochs.')
         logger.info(gen_log_message(message=msg, step=4, subject=subject,
                                     session=session))
+        if not eog_inds:
+            warn = ('No EOG-related ICs detected, '
+                    'this is highly suspicious. '
+                    'A manual check is suggested. '
+                    'You can either lower "ica_eog_threshold" or check '
+                    'that the blinks are between the tasks '
+                    ' and not during the tasks.')
```
Comment (hoechenberger) [inline scripts/preprocessing/04a-run_ica.py:220]:
```suggestion
                    'and not during the tasks.')
```

File: scripts/preprocessing/04a-run_ica.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (crsegerie) [reply scripts/preprocessing/04a-run_ica.py]: But you are right, that's not clear. the commit "change the phrasing" should solve this. Tell me if it's more clear ?

**Comment by crsegerie** [comment]
Thanks, Alexandre and Richard for the help!

**Comment by hoechenberger** [comment]
Thanks @crsegerie 🚀

**Pull request merged**